onnxruntime/core/framework/graph_partitioner.cc

```diff
@@ -727,7 +729,20 @@ static Status CreateEpContextModel(const ExecutionProviders& execution_providers
     }
   }
 
-  ORT_RETURN_IF_ERROR(Model::Save(ep_context_model, context_cache_path));
+  size_t ini_size_threshold = 0;
+  std::filesystem::path external_ini_path;
+  if (ep_context_ext_ini_path.empty()) {
+    // Set the threshold to the max so all initializers are forced into the Onnx file
+    ini_size_threshold = SIZE_MAX;
+    external_ini_path = "./model_ext_ini.bin";
```

Comment:
I'm assuming this a throwaway file path. Is it required to have some non-empty value before calling `Model::SaveWithExternalInitializers()`?

If so, could consider calling the normal `Model::Save()` when the `ep_context_ext_ini_path` is empty. Just a minor suggestion.

Reply:
Same as I replied in another comment, SaveWithExternalInitializers() creates the file first, so it doesn't allow empty path string. And Model::Save has issue with existing external data, it do nothing to external data.